src/ape/managers/compilers.py

```diff
@@ -116,6 +116,61 @@ def compile(self, contract_filepaths: List[Path]) -> Dict[str, ContractType]:
 
         return contract_types_dict  # type: ignore
 
+    def get_imports(
+        self, contract_filepaths: List[Path], base_path: Optional[Path]
+    ) -> Dict[str, List[str]]:
+        """
+        Combines import dicts from all compilers, where the key is a contract's source_id
+        and the value is a list of import source_ids.
+
+        Args:
+            contract_filepaths (List[pathlib.Path]): A list of source file paths to compile.
+            base_path (Optional[pathlib.Path]): Optionally provide the base path, such as the
+              project ``contracts/`` directory. Defaults to ``None``. When using in a project
+              via ``ape compile``, gets set to the project's ``contracts/`` directory.
+
+        Returns:
+            Dict[str, List[str]] {source_id: [import_source_id, ...], ...}
+        """
+        imports_dict: Dict[str, List[str]] = {}
+
+        for _, compiler in self.registered_compilers.items():
+            try:
+                imports = compiler.get_imports(
+                    contract_filepaths=contract_filepaths, base_path=base_path
+                )
+            except NotImplementedError:
+                imports = None
+
+            if imports:
+                imports_dict.update(imports)
+
+        return imports_dict
+
+    def get_references(self, imports_dict: Dict[str, List[str]]) -> Dict[str, List[str]]:
+        """
+        Reversed dictionary from contract imports dictionary. Each entry is a list of
+        source_ids that reference the source_id key of a given contract.
+
+        Args:
+            contract_filepaths (List[pathlib.Path]): A list of source file paths to compile.
+            base_path (Optional[pathlib.Path]): Optionally provide the base path, such as the
+              project ``contracts/`` directory. Defaults to ``None``. When using in a project
+              via ``ape compile``, gets set to the project's ``contracts/`` directory.
+
+        Returns:
+            Dict[str, List[str]] {source_id: [referring_source_id, ...], ...}
+        """
+        references_dict: Dict[str, List[str]] = {}
+        if imports_dict:
```

Comment:
Optional nit, if you reverse the `if` statement, you could de-indent most of this method.

```python
if not imports_dict:
    return {}

for key, imports_list in imports_dict.items():
    ...
```

Reply:
Updated [b8b78ed](https://github.com/ApeWorX/ape/pull/730/commits/b8b78ed1cdf6910721bd7cb4b168dfae1bc1f1f4)